lib/components/intent.js

```diff
@@ -48,12 +48,16 @@ var STATE_EVENT_TYPES = [
  * @param {boolean} opts.dontJoin True to not attempt to join a room before
  * sending messages into it. The surrounding code will have to ensure the correct
  * membership state itself in this case. Default: false.
+ * 
+ * @param {boolean} [opts.enablePresence=true] True to send presence, false to no-op.
  */
 function Intent(client, botClient, opts) {
     this.client = client;
     this.botClient = botClient;
     opts = opts || {};
 
+    opts.enablePresence = !(opts.enablePresence !== false);
```

Comment:
or `!!opts.enablePresence`? I assume you're trying to achieve a default true unless false here, which a double negative can do.

Reply:
Feelin more dumb now :). I'll update.